gunpowder-api/src/main/kotlin/io/github/gunpowder/api/util/Location.kt

```diff
@@ -32,18 +32,22 @@ import net.minecraft.util.math.Vec2f
 import net.minecraft.util.math.Vec3d
 import net.minecraft.util.registry.Registry
 import net.minecraft.util.registry.RegistryKey
+import net.minecraft.world.World
 
 data class Location(
-        val position: Vec3d,
-        val rotation: Vec2f,
-        val dimension: Identifier
+        var position: Vec3d,
```

Comment:
These should not be mutable. Make functions to offset or change them into new objects.